### PR TITLE
Implementation of the PUT method and updating of data to the database…

### PR DESCRIPTION
**Pull request by AlejandroG1997**
…. data conversion implementation with orika.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -36,6 +36,30 @@
 			<artifactId>spring-boot-starter-data-mongodb</artifactId>
 		</dependency>
 		
+		<dependency>
+			<groupId>de.flapdoodle.embed</groupId>
+			<artifactId>de.flapdoodle.embed.mongo</artifactId>
+			<scope>test</scope>
+		</dependency>
```
Comment (soydianapinto) [inline pom.xml]: Why are you adding this dependency?

File: pom.xml [hunk 2]
```diff
@@ -36,6 +36,30 @@
...
+		
+		<dependency>
+			<groupId>net.rakugakibox.spring.boot</groupId>
+			<artifactId>orika-spring-boot-starter</artifactId>
+			<version>1.9.0</version>
+		</dependency>
```
Comment (soydianapinto) [inline pom.xml]: Why are you adding this dependency?

File: pom.xml [hunk 3]
```diff
@@ -36,6 +36,30 @@
...
+				<groupId>io.springfox</groupId>
+				<artifactId>springfox-swagger2</artifactId>
+				<version>2.9.2</version>
+			</dependency>
+
+			<dependency>
+				<groupId>io.springfox</groupId>
+				<artifactId>springfox-swagger-ui</artifactId>
+				<version>2.9.2</version>
+			</dependency>
```
Comment (soydianapinto) [inline pom.xml]: There's no Swagger implementation for this sprint.

File: src/main/java/com/stories/ApiApplication.java [hunk 4]
```diff
@@ -6,7 +6,7 @@
 @SpringBootApplication
 public class ApiApplication {
 
-	public static void main(String[] args) {
+	public static void main(String[] args) {	
 		SpringApplication.run(ApiApplication.class, args);
```
Comment (soydianapinto) [inline src/main/java/com/stories/ApiApplication.java]: Please replace "ApiApplication" to "StoriesApplication".

File: src/main/java/com/stories/domain/StoryDomain.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import lombok.Setter;
+import lombok.ToString;
+
+
+
+//@Setter
+//@Getter
+//@AllArgsConstructor
+//@NoArgsConstructor
+//@ToString
```
Comment (soydianapinto) [inline src/main/java/com/stories/domain/StoryDomain.java]: Please delete this comments.

File: src/main/java/com/stories/mapper/StoryMapping.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+//implements OrikaMapperFactoryConfigurer
+public class StoryMapping implements OrikaMapperFactoryConfigurer {
+
+	@Override
+	public void configure(MapperFactory orikaMapperFactory){
+		orikaMapperFactory.classMap(StoryDomain.class, StoryModel.class).byDefault().register();
+	}
+
+}
```
Comment (soydianapinto) [inline src/main/java/com/stories/mapper/StoryMapping.java]: Where are you using this StoryMapping class in your application?

File: src/main/java/com/stories/model/Story.java [hunk 7]
```diff
@@ -1,8 +0,0 @@
-package com.stories.model;
-
-import lombok.Data;
-
-@Data
-public class Story {
-	
-}
```
Comment (soydianapinto) [inline src/main/java/com/stories/model/Story.java]: Use this file instead of deleting it.

File: src/main/java/com/stories/model/StoryModel.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    private Date due_date;
+    private String priority;
+    private String assignee_id;
+    private ArrayList history;
+	public String get_id() {
+		return _id;
+	}
+	public void set_id(String _id) {
+		this._id = _id;
+	}
```
Comment (soydianapinto) [inline src/main/java/com/stories/model/StoryModel.java]: Why are you using this?

File: src/main/java/com/stories/repository/StoriesRepository.java [hunk 9]
```diff
@@ -1,9 +1,13 @@
 package com.stories.repository;
 
+
```
Comment (soydianapinto) [inline src/main/java/com/stories/repository/StoriesRepository.java]: Please delete this space.

File: src/main/java/com/stories/services/StoryService.java [hunk 10]
```diff
@@ -0,0 +1,9 @@
+package com.stories.services;
+
+import com.stories.domain.StoryDomain;
+
+public interface StoryService {
+//	createStoryRequest request,
```
Comment (soydianapinto) [inline src/main/java/com/stories/services/StoryService.java]: Please delete this comment.

File: src/main/java/com/stories/services/StoryServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+
+import com.stories.domain.StoryDomain;
+import com.stories.model.StoryModel;
+import com.stories.repository.StoriesRepository;
+import ma.glasnost.orika.MapperFacade;
+import ma.glasnost.orika.MapperFactory;
+
+
```
Comment (soydianapinto) [inline src/main/java/com/stories/services/StoryServiceImpl.java]: Please delete this space.

File: src/main/java/com/stories/services/StoryServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+@Service
+public class StoryServiceImpl {
+	
+	@Autowired
+	 StoriesRepository storiesrepository;
```
Comment (soydianapinto) [inline src/main/java/com/stories/services/StoryServiceImpl.java]: Please use the camelCase practice for this instance.

File: src/main/java/com/stories/services/StoryServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+	
+	@Autowired
+	private MapperFactory orikaMapperFacade;
+
```
Comment (soydianapinto) [inline src/main/java/com/stories/services/StoryServiceImpl.java]: Please delete this space.

File: src/main/java/com/stories/swaggerConfig/Config.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+	
+	@Bean
+	public Docket SwaggerApi() {
+		return new Docket(DocumentationType.SWAGGER_2)
+				.select()
+				.apis(RequestHandlerSelectors.basePackage("com.stories"))
+				.paths(PathSelectors.any())
+				.build();
+	}
+}
```
Comment (soydianapinto) [inline src/main/java/com/stories/swaggerConfig/Config.java]: Why are you adding Swagger in this sprint?

File: src/main/resources/application.properties [hunk 15]
```diff
@@ -1,2 +1,13 @@
...
+#pass: vloUr7AakeTEdXDj
+
+spring.data.mongodb.uri=mongodb+srv://db_user:vloUr7AakeTEdXDj@cluster0-nomfh.mongodb.net/InternHome
+#spring.data.mongodb.uri=mongodb+srv://Ale_dbuser:Alejandro@cluster0-sas3p.mongodb.net/test?retryWrites=true&w=majority/intern
+
+#spring.data.mongodb.uri=mongodb+srv://Ale_dbuser:intern@cluster0-sas3p.mongodb.net/intern
+#spring.data.mongodb.uri=mongodb+srv://Ale_dbuser:intern@cluster0-sas3p.mongodb.net/test?retryWrites=true&w=majority/intern
+spring.data.mongodb.database = InternHome
+
+#5e6290031c9d44000033734d~
```
Comment (soydianapinto) [inline src/main/resources/application.properties]: Please delete all unnecessary properties and replace the password to < password >.

**Review by soydianapinto — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (AlejandroG1997) [reply pom.xml]: because I believed that this dependency was necessary for the junits, but this is not necessary

Comment (soydianapinto) [reply pom.xml]: Yes, I agree, please delete this dependency if you're not using it.

File: pom.xml (hunk 2 above)
Comment (AlejandroG1997) [reply pom.xml]: For convert the object model to the domain

File: src/main/java/com/stories/mapper/StoryMapping.java (hunk 6 above)
Comment (AlejandroG1997) [reply src/main/java/com/stories/mapper/StoryMapping.java]: on the controller, but this not working.

File: src/main/java/com/stories/model/StoryModel.java (hunk 8 above)
Comment (AlejandroG1997) [reply src/main/java/com/stories/model/StoryModel.java]: because I couldn't use the setters and getters with Lombok

File: src/main/java/com/stories/swaggerConfig/Config.java (hunk 14 above)
Comment (AlejandroG1997) [reply src/main/java/com/stories/swaggerConfig/Config.java]: For try response of the PUT method

File: src/main/java/com/stories/mapper/StoryMapping.java (hunk 6 above)
Comment (soydianapinto) [reply src/main/java/com/stories/mapper/StoryMapping.java]: Please delete it if you're not using it.

File: pom.xml (hunk 2 above)
Comment (soydianapinto) [reply pom.xml]: This dependency is used by the StoryMapping class that you're not using, please delete it.

File: src/main/java/com/stories/swaggerConfig/Config.java (hunk 14 above)
Comment (soydianapinto) [reply src/main/java/com/stories/swaggerConfig/Config.java]: Swagger implementation is not required for this Sprint, please remove it.

File: src/main/java/com/stories/StoriesApplication.java [hunk 16]
```diff
@@ -0,0 +1,16 @@
+package com.stories;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ComponentScan;
+
+@SpringBootApplication
+@ComponentScan("com.stories")
```
Comment (IvanVegaS) [inline src/main/java/com/stories/StoriesApplication.java]: There's no need to add the package inside the @ComponentScan since it's the same as the package where this class is in.

File: src/main/java/com/stories/controller/StoriesController.java [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+
+@RestController
+@RequestMapping(value = "/stories", produces = "application/json")
+public class StoriesController {
+
+	@Autowired
+	StoryServiceImpl storyService;
+
+	@ResponseStatus(value = HttpStatus.OK)
+	@PutMapping("/StoryUpdate")
```
Comment (IvanVegaS) [inline src/main/java/com/stories/controller/StoriesController.java]: URLs should always be lowercase. This endpoint should not use StoryUpdate, it should expect an ID as path parameter.

File: src/main/java/com/stories/controller/StoriesController.java [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+import com.stories.domain.StoryDomain;
+
+@RestController
+@RequestMapping(value = "/stories", produces = "application/json")
+public class StoriesController {
+
+	@Autowired
+	StoryServiceImpl storyService;
+
+	@ResponseStatus(value = HttpStatus.OK)
```
Comment (IvanVegaS) [inline src/main/java/com/stories/controller/StoriesController.java]: Response status should be Accepted, not OK.

File: src/main/java/com/stories/controller/StoriesController.java [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+	@PutMapping("/StoryUpdate")
+	public StoryDomain updateStory(@RequestBody StoryDomain request,
+			@RequestParam(value = "StoryId", defaultValue = "") String id) {
```
Comment (IvanVegaS) [inline src/main/java/com/stories/controller/StoriesController.java]: Remove RequestParam and use a Path Variable instead.

File: src/main/java/com/stories/domain/StoryDomain.java [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+	private String name;
+	private String description;
+	private String acceptance_criteria;
+	private int points;
+	private int progress;
+	private String status;
+	private String notes;
+	private String comments;
+	private Date start_date;
+	private Date due_date;
```
Comment (IvanVegaS) [inline src/main/java/com/stories/domain/StoryDomain.java]: **Always** use `LocalDate `or `LocalDateTime `instead of `Date`.

File: src/main/java/com/stories/domain/StoryDomain.java [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+	private String priority;
+	private String assignee_id;
+	private ArrayList history;
```
Comment (IvanVegaS) [inline src/main/java/com/stories/domain/StoryDomain.java]: Change `ArrayList `to `List `of String.

File: src/main/java/com/stories/domain/StoryDomain.java [hunk 22]
```diff
@@ -0,0 +1,26 @@
+package com.stories.domain;
+
+import java.util.ArrayList;
+import java.util.Date;
+import lombok.Data;
+
+@Data
+public class StoryDomain {
+
```
Comment (IvanVegaS) [inline src/main/java/com/stories/domain/StoryDomain.java:14]: Add no-args constructor to initialize history to a new `ArrayList `of tipe `String`.

File: src/main/java/com/stories/model/StoryModel.java [hunk 23]
```diff
@@ -0,0 +1,33 @@
+package com.stories.model;
+
```
Comment (IvanVegaS) [inline src/main/java/com/stories/model/StoryModel.java]: Same observations as in the **StoryDomain**.

File: src/main/resources/application.properties [hunk 24]
```diff
@@ -1,2 +1,3 @@
 server.port=5000
-spring.data.mongodb.uri=mongodb+srv://<username>:<password>@cluster0-nomfh.mongodb.net/InternHome
+spring.data.mongodb.uri=mongodb+srv://db_user:<password>@cluster0-nomfh.mongodb.net/InternHome
+spring.data.mongodb.database = InternHome
```
Comment (IvanVegaS) [inline src/main/resources/application.properties:3]: Why is this needed?

File: src/main/java/com/stories/services/StoryServiceImpl.java [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+@Service
+public class StoryServiceImpl {
+
+	@Autowired
+	StoriesRepository storiesRepository;
+
+	public StoryDomain updateStory(StoryDomain request, String id) {
+		if (!storiesRepository.existsById(id))
+			throw new HttpClientErrorException(HttpStatus.NOT_FOUND, "Story not found");
+		MapperFactory mapperFactory = new DefaultMapperFactory.Builder().build();
```
Comment (IvanVegaS) [inline src/main/java/com/stories/services/StoryServiceImpl.java]: Separate Mapper logic in a different class and inject the mapper factory bean.
See example [here](https://github.com/IvanVegaS/DnData/blob/develop/src/main/java/com/ivcoding/services/dndata/service/RacesServiceImpl.java)

**Review by IvanVegaS — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 24 above)
Comment (AlejandroG1997) [reply src/main/resources/application.properties:3]: because the URI is necessary for the connection to the database, but this section "/ InternHome" is not necessary.

**Review by soydianapinto — APPROVED** [review]

**Review by at-internship — APPROVED** [review]
